### PR TITLE
Enable nullable: System.Management.Automation.Internal.IValidateSetValuesGenerator

### PR DESCRIPTION
**Pull request by xtqqczze**
_Tracking issue: #12631._

### CODE REVIEW

File: src/System.Management.Automation/engine/Attributes.cs [hunk 1]
```diff
@@ -1731,13 +1731,15 @@ public ValidateSetAttribute(Type valuesGeneratorType)
     /// <summary>
     /// Allows dynamically generate set of values for <see cref="ValidateSetAttribute"/>
     /// </summary>
+#nullable enable
```
Comment (xtqqczze) [inline src/System.Management.Automation/engine/Attributes.cs:1734]: @iSazonov What is our code convention for indent of pre-processor directives. Visual Studio 2019 is automatically removing any indent here.

File: src/System.Management.Automation/engine/Attributes.cs [hunk 2]
```diff
@@ -1731,13 +1731,15 @@ public ValidateSetAttribute(Type valuesGeneratorType)
...
     public interface IValidateSetValuesGenerator
     {
         /// <summary>
         /// Gets valid values.
         /// </summary>
         string[] GetValidValues();
     }
+#nullable restore
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/Attributes.cs:1742]: Should it be:
```c#
string?[] GetValidValues();
```
We have internal check and throw if the method returns null but I don't see that we check the strings is not null. 
Thoughts?

File: src/System.Management.Automation/engine/Attributes.cs (hunk 1 above)
Comment (xtqqczze) [reply src/System.Management.Automation/engine/Attributes.cs:1734]: Related issue: https://github.com/dotnet/roslyn/issues/12306

Comment (iSazonov) [reply src/System.Management.Automation/engine/Attributes.cs:1734]: If we can not to turn off the behavior in VS 2019 I agree to put this at the beginning of the line.

File: src/System.Management.Automation/engine/Attributes.cs (hunk 2 above)
Comment (xtqqczze) [reply src/System.Management.Automation/engine/Attributes.cs:1742]: The PR builds and passes tests as-is and this is an internal API.

Comment (xtqqczze) [reply src/System.Management.Automation/engine/Attributes.cs:1742]: It doesn't appear this API was designed to allow `null` as a "valid value", but I can't know the design intent for certain.

Comment (iSazonov) [reply src/System.Management.Automation/engine/Attributes.cs:1742]: /cc @SeeminglyScience  @vexx32

Comment (iSazonov) [reply src/System.Management.Automation/engine/Attributes.cs:1742]: In ValidateSetAttribute class we check the element in ValidateElement() method and throw if it is null.

@xtqqczze Please enhance the XML comment for GetValidValues() with the two design intentions:
- return is not null
- elements is not null

Comment (vexx32) [reply src/System.Management.Automation/engine/Attributes.cs:1742]: Generally if PS encounters $null for what is a string it'll convert it to `string.Empty`, so unless folks are constructing a values generator and handing back `[nullstring]::Value` it _shouldn't_ be a concern for most things, I wouldn't think?

Comment (iSazonov) [reply src/System.Management.Automation/engine/Attributes.cs:1742]: In C# applications no conversion to `[nullstring]::Value` exists and ValidateElement() will throws on null element.

Comment (vexx32) [reply src/System.Management.Automation/engine/Attributes.cs:1742]: I wouldn't generally expect null elements to be supported anyway tbh, it kind of defeats the purpose of the values generator if it's handing back null values in the first place?

Comment (iSazonov) [reply src/System.Management.Automation/engine/Attributes.cs:1742]: Yes, current behavior is right.

**Review by rjmholt — APPROVED** [review]

**Comment by xtqqczze** [comment]
Added documentation comment: `<returns>A non-null array of non-null strings.</returns>`

**Review by iSazonov — APPROVED** [review]

**Comment by iSazonov** [comment]
Great start! :-)

**Pull request merged**